assets/src/components/ui/toggle-button/toggle-button.js

```diff
@@ -0,0 +1,40 @@
+/**
+ * ToggleButton
+ * Returns a component that can be used to trigger an action like opening or closing a container.
+ * The component appears as an icon only but acts like a button.
+ *
+ * @function
+ * @param {string} htmlId               Used to generate an HTML "id" attribute for all components.
+ *                                      Additional text is appended by each component.
+ * @param {string} htmlClass            Used to generate an HTML "class" attribute for all components.
+ *                                      Additional text is appended by each component.
+ * @param {string} description          Two or three words (hyphenated) that describe the container
+ *                                      that the ToggleButton is associated with.
+ * @param {Function} toggleCallback 	Function called when the "button" component is triggered
+ * @param {string} dashicon             Css class that defines the dashicon icon
+ * @param {Number} tabIndex             tab index for the element that controls when it receives focus
+ * @return {Object}                     "button" for opening the modal container
+ */
+export const ToggleButton = (
+	{
+		htmlId,
+		htmlClass,
+		description,
+		toggleCallback,
+		dashicon = 'no',
+		tabIndex = 0,
+	}
+) => {
+	return toggleCallback && (
+		<div
+			id={ `${ htmlId }-${ description }` }
```

Comment:
I think this should still be wrapped by the `withInstanceId` HOC to make sure this is unique.

Reply:
As explained in #597 you do not need to wrap absolutely everything that sets an HTML id with the `withInstanceId` HOC. That would be total overkill. There is nothing stopping this id from being unique, but the dev would have to do so manually, which could include an id that has been passed down through the app that was partially generated using the `withInstanceId` HOC.